utils/logging/log_test.go

```diff
@@ -23,3 +38,39 @@ func TestLog(t *testing.T) {
 
 	require.True(t, *recovered)
 }
+
+func TestWith(t *testing.T) {
+	testWriter := &testLogWriter{}
+	parentLogger := NewLogger("", NewWrappedCore(Info, testWriter, Plain.ConsoleEncoder()))
+
+	childLogger := parentLogger.With(zap.Bool("bool", true))
+	childLogger.Info("child message")
+	require.Len(t, testWriter.logs, 1)
+	require.Contains(t, string(testWriter.logs[0]), `"bool": true`)
+
+	parentLogger.Info("parent message")
+	require.Len(t, testWriter.logs, 2)
+	require.NotContains(t, string(testWriter.logs[1]), `"bool": true`)
+}
+
+func TestWithOption(t *testing.T) {
+	testWriter := &testLogWriter{}
+	parentLogger := NewLogger("", NewWrappedCore(Info, testWriter, Plain.ConsoleEncoder()))
+
+	hookTriggered := new(bool)
```

Comment:
you don't need to have a pointer, you can have a regular bool.

Reply:
Indeed -- updated.